### PR TITLE
Added lock button to switch between edit/read-only mode

### LINKED ISSUE #7: Add a lock snippet button that locks the snippet into place
This would stop the parsing and enter "Annotation Mode"

### PR DESCRIPTION
**Pull request by solkaz**
Fixes #7

### CODE REVIEW

**Comment by thebho** [comment]
Can you provide a screenshot?

File: src/components/AppBody.jsx [hunk 1]
```diff
@@ -44,6 +47,12 @@ class AppBody extends React.Component {
     });
   }
 
+  switchReadOnlyMode() {
+    this.setState({
+      readOnly: !(this.state.readOnly),
```
Comment (dane-johnson) [inline src/components/AppBody.jsx]: I think this should be one way. You lock it in, it's locked in for good.

File: src/components/LockButton.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { PropTypes } from 'react';
+
+import FlatButton from 'material-ui/FlatButton';
+
+
+const LockButton = ({onClick, readOnly}) => {
+  return (
+    <FlatButton
+      label={`Switch to ${ readOnly ? 'edit' : 'read-only' } mode`}
```
Comment (dane-johnson) [inline src/components/LockButton.jsx]: Which would make this disappear when locked in.

File: src/components/AppBody.jsx (hunk 1 above)
Comment (thebho) [inline src/components/AppBody.jsx]: Right, there is no "edit snippet" path after locking in this iteration

**Pull request merged**